pkg/operator/selector_eventhandler_test.go

```diff
@@ -1,12 +1,11 @@
 // These tests depend on test assets from controller-runtime which don't work on Windows.
 
-// +build !windows
+// +build !windows,has_network
```

Comment:
This seems to execute despite me not setting `has_network` during a build…

Reply:
This can be reproduced by running `nix-build -A grafana-agent` from a checkout of https://github.com/flokli/nixpkgs, `grafana-agent-remove-patch` branch.

Reply:
🤔 That's strange, I don't see the same behavior:

```
$ go test -v ./pkg/operator
?       github.com/grafana/agent/pkg/operator   [no test files]
$ go test -tags=has_network -v ./pkg/operator
=== RUN   TestEnqueueRequestForSelector
# ... 
```

What version of Go are you using?

Reply:
(We had to change this because of [this comment](https://github.com/grafana/agent/pull/773#discussion_r687958362): turns out I was wrong about how the flags passing worked and the build tags are the only way to do this sanely, otherwise we'd need to add support for the network flag in every package)

Reply:
That's `go 1.16.7`. If you grok at https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/go-modules/generic/default.nix#L168, that's how we invoke the build.

Reply:
Ah, thanks, I was struggling to find that. 

I see what the problem is now: v0.18.2 is a cherry-picked release containing neither your flag change or this PR: https://github.com/grafana/agent/blob/v0.18.2/pkg/operator/selector_eventhandler_test.go. 

We've been having to do a few cherry-pick releases since main contains unfinished work that we can't release yet. We thinks it's confusing (leads to situations like this) and also it's a pain to manage. We're currently working on unblocking main and coming up with a better defined branching and release strategy to avoid getting in this situation again. The current plan is to have a full release of main prior to the 10th of September.

Reply:
Hm, I can reproduce the commands above do the same. It might be something specific to our test phase - we might not honor tags there properly. I'll check back on this, sorry for the noise.

Reply:
Well, apparently this landed in main after the `v0.18.2` tag, so I can debug this all day :facepalm:

Reply:
Okay, I can confirm this works just fine on current main. Sorry for the noise.

Reply:
No worries, sorry for the confusion with v0.18.2!